### PR TITLE
Fix race condition when updating tags

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: app/commands/exercise/update_tags.rb [hunk 1]
```diff
@@ -12,6 +12,8 @@ def tags
 
     new_tags = (solution_tags - exercise_tags).map do |tag|
       Exercise::Tag.find_or_create_by!(tag:, exercise:)
```
Comment (iHiD) [inline app/commands/exercise/update_tags.rb:14]:
```suggestion
      Exercise::Tag.find_create_or_find_by!!(tag:, exercise:)
```

File: app/commands/exercise/update_tags.rb [hunk 2]
```diff
@@ -12,6 +12,8 @@ def tags
...
+    rescue ActiveRecord::RecordNotUnique
+      Exercise::Tag.find_by!(tag:, exercise:)
```
Comment (iHiD) [inline app/commands/exercise/update_tags.rb]:
```suggestion
      Exercise::Tag.find_create_or_find_by!(tag:, exercise:)
```

Comment (ErikSchierboom) [reply app/commands/exercise/update_tags.rb]: Did you mean to suggest this for the original line, not for the exception?

Comment (iHiD) [reply app/commands/exercise/update_tags.rb]: I think this replaces all three lines.

Comment (iHiD) [reply app/commands/exercise/update_tags.rb]: The `or_find_by!` does exactly this under the hood.

Comment (ErikSchierboom) [reply app/commands/exercise/update_tags.rb]: Okay good!

**Pull request merged**